src/ripple/app/tx/impl/DeleteAccount.cpp

```diff
@@ -214,6 +214,23 @@ DeleteAccount::preclaim(PreclaimContext const& ctx)
     if ((*sleAccount)[sfSequence] + seqDelta > ctx.view.seq())
         return tecTOO_SOON;
 
+    // When fixNFTokenRemint is enabled, we don't allow an account to be
+    // deleted if <FirstNFTokenSequence + MintedNFTokens> is within 256 of the
+    // current ledger. This is to prevent having duplicate NFTokenIDs after
+    // account re-creation.
+    //
+    // Without this restriction, duplicate NFTokenIDs can be reproduced when
+    // authorized minting is involved. Because when the minter mints a NFToken,
+    // the issuer's sequence does not change. So when the issuer re-creates
+    // their account and mints a NFToken, it is possible that the
+    // NFTokenSequence of this NFToken is the same as the one that the
+    // authorized minter minted in a previous ledger.
+    if (ctx.view.rules().enabled(fixNFTokenRemint) &&
+        ((*sleAccount)[~sfFirstNFTokenSequence].value_or(0) +
+             (*sleAccount)[~sfMintedNFTokens].value_or(0) + seqDelta >
+         ctx.view.seq()))
+        return tecTOO_SOON;
```

Comment:
Imagine someone issuing a million "Silly Billy" NFTs, with the first minted in ledger 77,777,777.

This check will prevent the issuer from deleting the account until, at the earliest, ledger 78,778,032; that's not gonna happen for around 4 million seconds, or around a month and a half.

Reply:
Yes, unfortunately this check needs to be in. Without it, the scenario below produces duplicate NFTID:
1. Alice's account sequence is at 1 
2. Bob is Alice's authorized minter
3. Bob mints 500 NFTs for Alice (NFTs will have sequences 1 - 501, since NFT sequence is computed by `FirstNFTokenSequence` + `MintedNFTokens`)
4. Alice deletes her account at ledger 257 (enforced by AccountDelete amendment)
5. Alice re-creates her account at ledger 258
6. Alice mints a NFT. `FirstNFTokenSequence` initialized to her account sequence (258), and MintedNFTokens to 0. This newly minted NFT would have a sequence number of 258, which is a duplicate of what she issued through authorized minting.
7.

Reply:
I understand. My point was that this kind of test means it's hard for someone to know _when_ their account can be deleted and, depending on the number of NFTs they issue. For some accounts (those which don't have `FirstNFTokenSequence` set) the answer is "immediately" and for others it's "well, sometime in the future."

Reply:
But I don't think this will be a common issue, since this restriction is only applicable for authorized minting situations. 

If the issuer has never issued an NFT through authorized minting, then, `FirstNFTokenSequence` + `MintedNFTokens` <= `AccountRoot Sequence` will always be true, which means the AccountDelete amendment (ledger > Account Sequence + 256) already contains the new restriction.